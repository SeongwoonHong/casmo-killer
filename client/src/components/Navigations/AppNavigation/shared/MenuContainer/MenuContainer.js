import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import './MenuContainer.scss';

const UserMenu = (props) => {

  return (
    <div className={ classnames(`Menu-container ${props.className}`, {
      active: props.active
    })}>

      <div className="menu-header nav-headers">
        <button
          type="button"
          className="top-nav-btn"
          onClick={ () => props.onClose() }>
          <i className="material-icons">
            close
          </i>
        </button>
        <h1>
          <Link to="/">
            CK BOARD
          </Link>
        </h1>
      </div>

      <div className="menu-body">
        { props.children }
      </div>

    </div>
  );

};

export default UserMenu;
