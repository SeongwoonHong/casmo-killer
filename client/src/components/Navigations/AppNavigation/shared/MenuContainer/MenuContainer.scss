.Menu-container {
    @include abs-elem(0, auto, auto, auto, fixed);
    height: 100%;
    transition: all .3s;
    width: $main-menu-width;
    z-index: 88;
    &.active {
        transform: translateX(0);
    }
    .menu-header {
        @include flex-box();
    }
    .menu-body {
        background-color: palette(Grey, 300);
        height: 100%;
        ul {
            list-style: none;
            > li {
                > a {
                    @include flex-box();
                    color: palette(Grey, 800);
                    font-size: 12px;
                    height: $top-nav-size;
                    overflow: hidden;
                    text-decoration: none;
                }
            }
        }
    }
    .main-menu {
        > li {
            > a {
                border-bottom: 1px solid transparent;
                border-top: 1px solid transparent;
                cursor: pointer;
                padding: 0 24px;
                &.active {
                    background-color: #fff;
                    border-color: $main-menu-border-color;
                    color: palette(Grey, 900);
                    font-weight: 700;
                    > .material-icons {
                        color: inherit;
                    }
                }
                > .material-icons {
                    color: #888888;
                    font-size: 20px;
                    height: $top-nav-size;
                    line-height: $top-nav-size;
                    margin-right: 27px;
                }
                > span {
                    line-height: $top-nav-size;
                }
            }
            &:first-child {
                > a {
                    &.active {
                        border-top: none;
                    }
                }
            }
        }
    }
    @media screen and (min-width: 992px) {
        height: auto;
        width: 100%;
        position: sticky;
        transition: none;
        transform: translateX(0) !important;
        .menu-header {
            display: none;
        }
        .menu-body {
            background-color: palette(Grey, 800);
            width: 100%;
        }
        .main-menu {
            @include flex-box(flex-start);
            margin: 0 auto;
            max-width: 960px;
        }
    }
}
