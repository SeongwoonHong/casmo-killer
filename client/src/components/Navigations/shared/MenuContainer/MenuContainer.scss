.Menu-container {
    @include abs-elem(0, auto, 0, auto, fixed);
    transition: all .3s;
    width: $main-menu-width;
    z-index: 88;
    &.active {
        transform: translateX(0);
    }
    &__header {
        @include flex-box();
        background-color: palette(Grey, 800);
        border-bottom: 1px solid palette(Grey, 900);
        height: $top-nav-size;
        .App-title__link {
            margin-left: 25px;
        }
    }
    &__body {
        background-color: palette(Grey, 300);
        height: 100%;
        ul {
            > li {
                > a {
                    @include flex-box();
                    height: $top-nav-size;
                    text-decoration: none;
                }
            }
        }
    }
    @media screen and (min-width: 992px) {
        width: 100%;
        position: sticky;
        transition: none;
        transform: translateX(0) !important;
        &__header {
            display: none;
        }
        &__body {
            background-color: palette(Grey, 800);
            width: 100%;
        }
        &__body {
            @include flex-box(flex-start);
            margin: 0 auto;
        }
    }
}
