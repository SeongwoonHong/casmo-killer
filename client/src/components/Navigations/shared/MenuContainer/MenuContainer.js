import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import TopNavBtn from '../TopNavBtn';
import AppTitle from '../AppTitle';

import './MenuContainer.scss';

const UserMenu = ({
  className, active, onClose, children
}) => {

  return (
    <div className={ classnames(`Menu-container ${className}`, {
      active
    })}>
      <div className="Menu-container__header">
        <TopNavBtn
          icon="close"
          classPrefix="Menu-container"
          onClick={ onClose } />
        <AppTitle />
      </div>
      <div className="Menu-container__body">
        { children }
      </div>

    </div>
  );

};

UserMenu.propTypes = {
  className: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.any.isRequired
};

export default UserMenu;
