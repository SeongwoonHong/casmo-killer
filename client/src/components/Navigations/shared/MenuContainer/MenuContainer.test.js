/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';

import MenuContainer from './index';

describe('<MenuContainer />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <MenuContainer
        active={ true }
        className="testing"
        onClose={ () => {} }>
        <p>testing</p>
      </MenuContainer>
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

});
