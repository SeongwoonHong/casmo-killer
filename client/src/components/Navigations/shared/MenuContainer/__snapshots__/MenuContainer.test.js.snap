// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuContainer /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserMenu
    active={true}
    className="testing"
    onClose={[Function]}
>
    <p>
        testing
    </p>
</UserMenu>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Menu-container__header"
>
          <TopNavBtn
                    className=""
                    classPrefix="Menu-container"
                    icon="close"
                    onClick={[Function]}
          />
          <AppTitle />
</div>,
        <div
          className="Menu-container__body"
>
          <p>
                    testing
          </p>
</div>,
      ],
      "className": "Menu-container testing active",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TopNavBtn
              className=""
              classPrefix="Menu-container"
              icon="close"
              onClick={[Function]}
/>,
            <AppTitle />,
          ],
          "className": "Menu-container__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "className": "",
              "classPrefix": "Menu-container",
              "icon": "close",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            testing
</p>,
          "className": "Menu-container__body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "testing",
          },
          "ref": null,
          "rendered": "testing",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Menu-container__header"
>
            <TopNavBtn
                        className=""
                        classPrefix="Menu-container"
                        icon="close"
                        onClick={[Function]}
            />
            <AppTitle />
</div>,
          <div
            className="Menu-container__body"
>
            <p>
                        testing
            </p>
</div>,
        ],
        "className": "Menu-container testing active",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TopNavBtn
                className=""
                classPrefix="Menu-container"
                icon="close"
                onClick={[Function]}
/>,
              <AppTitle />,
            ],
            "className": "Menu-container__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
                "className": "",
                "classPrefix": "Menu-container",
                "icon": "close",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              testing
</p>,
            "className": "Menu-container__body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testing",
            },
            "ref": null,
            "rendered": "testing",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
